SOOUM/SOOUM/Presentations/Onboarding/NicknameSetting/OnboardingNicknameSettingViewController.swift

```diff
@@ -0,0 +1,221 @@
+    }
+    
+    private let errorLogStackView = UIStackView().then {
+        $0.axis = .horizontal
+        $0.spacing = 4
+        $0.alignment = .center
+        
+        let imageView = UIImageView().then {
+            $0.image = .error
+            $0.contentMode = .scaleAspectFit
+        }
+        $0.addArrangedSubviews(imageView)
+        $0.isHidden = true
+    }
+    
+    private let errorLogLabel = UILabel().then {
+        $0.typography = .init(
+            fontContainer: BuiltInFont(
+                size: 14,
+                weight: .regular
+            ),
+            lineHeight: 19.6,
+            letterSpacing: 0
+        )
+        // TODO: - 색 수정 필요
+        $0.textColor = .som.red
+        $0.text = "한 글자 이상 입력해주세요"
+    }
+    
+    private let nicknameCountLabel = UILabel().then {
+        $0.typography = .init(
+            fontContainer: BuiltInFont(
+                size: 14,
+                weight: .regular
+            ),
+            lineHeight: 19.6,
+            letterSpacing: 0
+        )
+        // TODO: - 색 수정 필요
+        $0.textColor = .som.gray500
+        $0.text = "1/8"
+    }
+    
+    private let nextButtonView = PrimaryButtonView()
+    
+    // Reactor를 연결하고, 액션과 상태를 바인딩합니다.
+    func bind(reactor: OnboardingNicknameSettingViewReactor) {
+        
+        nicknameTextField.textField.rx.text.orEmpty
+            .observe(on: MainScheduler.instance)
+            .subscribe(onNext: { [weak self] str in
+                guard let self = self else { return }
+                self.nicknameCountLabel.text = "\(str.count)/\(self.maxCount)"
+            })
+            .disposed(by: disposeBag)
+        
+        nicknameTextField.textField.rx.text.orEmpty
+            .debounce(.milliseconds(300), scheduler: MainScheduler.instance)
+            .distinctUntilChanged()
+            .map { Reactor.Action.textChanged($0) }
+            .bind(to: reactor.action)
+            .disposed(by: disposeBag)
+        
+        nicknameTextField.clearButtonView.rx.tapGesture()
+            .when(.recognized)
+            .subscribe(with: self) { object, _ in
+                object.nicknameCountLabel.text = "0/\(object.maxCount)"
+                object.nicknameTextField.textField.text?.removeAll()
+                object.reactor?.action.onNext(.textChanged(""))
+            }
+            .disposed(by: disposeBag)
+        
+        nextButtonView.rx.tapGesture()
+            .when(.recognized)
+            .subscribe(with: self) { object, _ in
+                let profileImageVC = ProfileImageSettingViewController()
+                let profileImageReactor = ProfileImageSettingViewReactor(nickname: self.nicknameTextField.textField.text!)
+                profileImageVC.reactor = profileImageReactor
+                self.navigationController?.pushViewController(profileImageVC, animated: true)
+            }
+            .disposed(by: disposeBag)
```

Comment:
```suggestion
        
        nextButtonView.rx.tapGesture()
            .when(.recognized)
            .subscribe(with: self) { object, _ in
                let profileImageVC = ProfileImageSettingViewController()
                let profileImageReactor = ProfileImageSettingViewReactor(nickname: object.nicknameTextField.textField.text!)
                profileImageVC.reactor = profileImageReactor
                object.navigationController?.pushViewController(profileImageVC, animated: true)
            }
            .disposed(by: disposeBag)
```
메모리 누수 방지를 위해 object를 사용해야 합니다.